cmds/cpio/types.go

```diff
@@ -0,0 +1,34 @@
+// Copyright 2013-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// basic types for cpio
+
+package main
+
+import "io"
+
+type RecReader interface {
+	RecRead() (*File, error)
+}
+
+type RecWriter interface {
+	RecWrite() (*File, error)
+}
+
+type File struct {
```

Comment:
Place a comment here saying parsing will break if the fields are changed or reordered.